### PR TITLE
[Java] Remove API `setJvmOptions(String)`.

### LINKED ISSUE #17391: [Java] Remove setJvmOptions(String) API totally.

### PR DESCRIPTION
**Pull request by jovany-wang**
This API has been deprecated for a while. It's time to remove it I believe.

User can use `setJvmOptions(List<String> jvmOptions)` instead.

Close #17391

### CODE REVIEW

**Comment by kfstorm** [comment]
I think we need approval from @pcmoritz for any API changes.

**Comment by jovany-wang** [comment]
@ericl

**Comment by jovany-wang** [comment]
Hi @kfstorm @raulchen @ericl ,
maybe Java API change needn't be approved by Anyscale guys?

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
Agree this isn't a core API change.

**Comment by jovany-wang** [comment]
@kfstorm Can you approve this? This PR should be approved by one of the codeowners.

**Review by kfstorm — APPROVED** [review]

**Pull request merged**
